server/neptune/workflows/activities/terraform/async_client.go

```diff
@@ -93,13 +98,55 @@ func (c *AsyncClient) RunCommand(ctx context.Context, request *RunCommandRequest
 		cmd.Env = append(cmd.Env, fmt.Sprintf("%s=%s", key, val))
 	}
 
-	if err := cmd.Run(); err != nil {
-		return errors.Wrap(err, "running terraform command")
+	if err := cmd.Start(); err != nil {
+		return errors.Wrap(err, "starting terraform command")
+	}
+
+	done := make(chan struct{})
+	defer close(done)
+	go terminateOnCtxCancellation(ctx, cmd.Process, done)
+
+	err = cmd.Wait()
+
+	if ctx.Err() != nil {
+		return errors.Wrap(ctx.Err(), "waiting for terraform process")
+	}
```

Comment:
when is this set?

Reply:
what do you mean? If ctx gets cancelled this will have an err

Reply:
oops, ignore did not mean to include this comment